components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java

```diff
@@ -651,8 +672,26 @@ public static Notification buildNotification(Event event, Map<String, String> pl
         }
 
         EmailTemplate emailTemplate;
+        String applicationUuid = null;
         try {
-            emailTemplate = NotificationHandlerDataHolder.getInstance().getEmailTemplateManager().getEmailTemplate(notificationEvent, locale, tenantDomain);
+            if (event.getEventProperties().containsKey("serviceProviderName")) {
+                String applicationName = event.getEventProperties().get("serviceProviderName").toString();
+                try {
+                    applicationUuid = NotificationHandlerDataHolder.getInstance().getApplicationManagementService()
+                            .getApplicationBasicInfoByName(applicationName, tenantDomain).getApplicationResourceId();
+                } catch (IdentityApplicationManagementException e) {
+                    // Fallback to organization preference if application is not found.
```

Comment:
We can add a WARN log.

Reply:
Addressed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/223/commits/43afd46f6a19a8fc329ccee7166d75003c866306